flocker/node/agents/ebs.py

```diff
@@ -609,7 +609,10 @@ def compute_instance_id(self):
         """
         Look up the EC2 instance ID for this node.
         """
-        return get_instance_metadata()['instance-id'].decode("ascii")
+        instance_id = get_instance_metadata().get('instance-id', None)
+        if not instance_id:
```

Comment:
Depends on what you mean by "possible". I have no evidence that it's occurred, and if it did, it would be a bug in AWS. However, it's just a streams of bits from some computer on the Internet, anything is possible. 

Having the check here helps by:
* catching the problem as soon as we're aware of it
* providing symmetry with other implementations

and harms by:
* increasing complexity of code in this function
* doing work up-front that might not be necessary

Truly, I don't know what's best. I guess I put it here because in other languages I use, in any lookup we'd be forced to locally handle the case of the lookup failing, which I think reduces mental overhead.